### PR TITLE
Issue #211 Making DNS host configuration a Vagrant host capabiltiy

### PR DESCRIPTION
**Pull request by hferentschik**

### CODE REVIEW

**Comment by hferentschik** [comment]
@budhrg @LalatenduMohanty @bexelbie - here is the pull request for automatic configuration of DNS on "known" Linux system. The overall change seems big, but that is mainly due to the fact that I did a quite big refactoring to make host DNS configuration a Vagrant host capability. Any feeback? Otherwise I'll go ahead and cut a 1.1.0.beta.2.

File: README.md [hunk 1]
```diff
@@ -177,7 +177,12 @@ through the host's VPN connection).
 ### Visibility on the Host
 
 Visibility on the host means that the hostname of the VMs can be resolved on the host's DNS system.
-Depending on the OS this might require some manual configuration.
+Landrush will attempt an automatic configuration of the host, but depending on the OS manual
```
Comment (budhrg) [inline README.md]: Need comma after `the OS` word.

**Comment by LalatenduMohanty** [comment]
@hferentschik +1 for configuring DNS as per host capability.

File: lib/landrush/cap/host/redhat/install_dnsmasq.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module Landrush
+  module Cap
+    module Redhat
+      module InstallDnsmasq
+        class << self
+          def install_dnsmasq(_env)
+            system('sudo yum clean all > /dev/null 2>&1')
+            system('sudo yum install -y -q dnsmasq > /dev/null 2>&1')
+          end
```
Comment (LalatenduMohanty) [inline lib/landrush/cap/host/redhat/install_dnsmasq.rb]: For Fedora we should use dnf. So we should add another host capability as Fedora which Vagrant provides us.

**Comment by LalatenduMohanty** [comment]
@hferentschik Which Linux distribution you have used to test the PR.

**Comment by LalatenduMohanty** [comment]
Also I do not see any documentation with the PR? or did I miss it?

File: lib/landrush/cap/host/redhat/install_dnsmasq.rb (hunk 2 above)
Comment (hferentschik) [inline lib/landrush/cap/host/redhat/install_dnsmasq.rb]: Happy to accept a pull request :-)

Comment (hferentschik) [inline lib/landrush/cap/host/redhat/install_dnsmasq.rb]: Vagrant does not provide host type check for Fedora out of the box. It does hardly for any OS. That's the reason why I had my own XYZHost classes.

File: README.md (hunk 1 above)
Comment (hferentschik) [inline README.md]: ok

**Comment by hferentschik** [comment]
@LalatenduMohanty 

> +1 for configuring DNS as per host capability. 

Yeah, makes the code more uniform and easy to add new host types

> Which Linux distribution you have used to test the PR.

Debian, Ubuntu, CentOS

> Also I do not see any documentation with the PR? or did I miss it? 

I think you did. Check the changes to the README

File: lib/landrush/cap/host/redhat/install_dnsmasq.rb (hunk 2 above)
Comment (LalatenduMohanty) [inline lib/landrush/cap/host/redhat/install_dnsmasq.rb]: @hferentschik checked Vagrant's code and you are right. I do not see code for Fedora in vagrant/plugins/hosts . This is crazy :(

**Pull request merged**